src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/MultiUserAKVTest.java

```diff
@@ -0,0 +1,643 @@
+            fail(TestResource.getResource("R_masterKeyPathNullOrEmpty"));
+        }
+
+        provider.setColumnEncryptionCacheTtl(Duration.ofSeconds(10));
+        byte[] plaintextKey = { 1, 2, 3 };
+        byte[] encryptedKey = provider.encryptColumnEncryptionKey(keyIDs[0], Constants.CEK_ALGORITHM, plaintextKey);
+
+        provider.decryptColumnEncryptionKey(keyIDs[0], Constants.CEK_ALGORITHM, encryptedKey);
+        assertTrue(cekCacheContainsKey(encryptedKey, provider));
+        assertEquals(1, getCacheSize(cekCacheSizeGetterName, provider));
+
+        Thread.sleep(TimeUnit.SECONDS.toMillis(11));
+        assertFalse(cekCacheContainsKey(encryptedKey, provider));
+        assertEquals(0, getCacheSize(cekCacheSizeGetterName, provider));
+    }
+
+    @Test
+    public void cekCacheShouldBeDisabledWhenAkvProviderIsRegisteredGlobally() throws Exception {
+        SQLServerColumnEncryptionAzureKeyVaultProvider provider = createAKVProvider();
+        
+        if (provider == null) {
+            fail(TestResource.getResource("R_AKVProviderNull"));
+        }
+
+        if (!isMasterKeyPathSetup) {
+            fail(TestResource.getResource("R_masterKeyPathNullOrEmpty"));
+        }
+
+        SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();
+        Map<String, SQLServerColumnEncryptionKeyStoreProvider> providerMap = new HashMap<String, SQLServerColumnEncryptionKeyStoreProvider>();
+        providerMap.put(Constants.AZURE_KEY_VAULT_NAME, provider);
+        SQLServerConnection.registerColumnEncryptionKeyStoreProviders(providerMap);
+
+        try (SQLServerConnection con = (SQLServerConnection) PrepUtil.getConnection(AETestConnectionString, AEInfo)) {
+            Field globalCustomCacheField = SQLServerConnection.class.getDeclaredField("globalCustomColumnEncryptionKeyStoreProviders");
+            globalCustomCacheField.setAccessible(true);
+
+            Map<String, SQLServerColumnEncryptionKeyStoreProvider> globalCacheFieldValue = (Map<String, SQLServerColumnEncryptionKeyStoreProvider>) globalCustomCacheField.get(con);
+
+            Method method = globalCacheFieldValue.getClass().getDeclaredMethod("get", Object.class);
+            SQLServerColumnEncryptionAzureKeyVaultProvider providerInGlobalCache = (SQLServerColumnEncryptionAzureKeyVaultProvider) method.invoke(globalCacheFieldValue, Constants.AZURE_KEY_VAULT_NAME);   
+
+            byte[] plaintextKey = { 1, 2, 3 };
+            byte[] encryptedKey = providerInGlobalCache.encryptColumnEncryptionKey(keyIDs[0], Constants.CEK_ALGORITHM, plaintextKey);
+
+            providerInGlobalCache.decryptColumnEncryptionKey(keyIDs[0], Constants.CEK_ALGORITHM, encryptedKey);
+            assertEquals(0, getCacheSize(cekCacheSizeGetterName, providerInGlobalCache));
+        } catch (SQLException e) {
+            fail(e.getMessage());
+        } finally {
+            // clean up
+            SQLServerConnection.unregisterColumnEncryptionKeyStoreProviders();

```

Comment:
do we need this? it's done in AfterAll

Reply:
This is to avoid conflict when running the following tests in the same file.